app/controllers/user.js

```diff
@@ -1,33 +1,112 @@
 const User = require('../models').users,
-  logger = require('../logger');
+  errors = require('../errors'),
+  logger = require('../logger'),
+  jwt = require('jwt-simple'),
+  secret = process.env.NODE_API_JWT_SECRET_STRING,
+  { check, validationResult } = require('express-validator/check'),
+  bcrypt = require('bcrypt');
+
+
+exports.signup = (req, res, next) => {
+
+  res.status(200);
+  
+  res.sendFile(path.resolve('app/views/user/signup.html'));
+
+};
 
 exports.create = (req, res, next) => {
 
-  const input = emptyToNull(req.body);
+  let input = emptyToNull(req.body);
 
   User.create({
     name: input.name,
     lastName: input.lastName,
     email: input.email,
     password: input.password
-  }).then(result => {
-    logger.info(`User ${input.name} created successfully.`);
-    res.status(201).send(result);
+  })
+    .then(result => {
+      let message = `User ${input.name} created successfully.`;
+      logger.info(message);
+      res.status(201).send({
+        name: result.name,
+        lastName: result.lastName,
+        email: result.email
+      });
 
-  }).catch(error => {
+    }).catch(error => {
+
+      let errorBag = [];
+      if(error.errors){
+        errorBag = error.errors.map(err => err.message);
+        logger.error(`A database error occured when attempting a user signup. Details: ${errorBag}.`);
+        res.status(401).send(errorBag);
+      }else{
+        logger.error(`Unhandled error! details: ${error}`);
+        res.status(500).send(error);
+      }
+
+    });
+
+  
+  
+};
+
+exports.signin = (req, res, next) => {
 
-    let errorBag = [];
-    if(error.errors){
-      errorBag = error.errors.map(err => err.message);
-      logger.error(`A database error occured when attempting a user signup. Details: ${errorBag}.`);
-      res.status(401).send(errorBag);
-    }else{
-      logger.error(`Unhandled error! details: ${error}`);
-      res.status(500).send(error);
+  try {
+    validationResult(req).throw();
+  } catch (err) {
+    return next(errors.invalidCredentialError);
+  }
+
+  let input = emptyToNull(req.body);
+
+  if (req.headers.token){
+
+    let token = jwt.decode(req.headers.token, secret);
+
+    if(token.token == input.email){
+      return res.status(200).send('You are already logged in!');
+    }
+    
+  }
+
+  User.findOne({
+    where: { email: input.email }
+  }).then(result => {
+
+    if(!result){
+      logger.info(`Failed login attempt to account with invalid email: "${input.email}"`);
+      return next(errors.invalidCredentialError);
     }
+    
+    bcrypt.compare(input.password, result.password).catch(err => {}).then(correctPassword => {
```

Comment:
If the compare fails, but are we returning? I think you can avoid that catch and let the error to propagate to the next catch that will handled the unexpected error.